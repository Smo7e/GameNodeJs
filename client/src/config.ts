// UdSU
// const DOMAIN = 'http://studfront';
// const PORT = 81;

// dev
const DOMAIN = "http://localhost";
const PORT = 3001;

// prod
//...

export const HOST = PORT ? `${DOMAIN}:${PORT}` : `${DOMAIN}`;

export const MEDIATOR = {
    EVENTS: {
        SERVER_ERROR: "SERVER_ERROR",
        GET_MESSAGES: "GET_MESSAGES",
        GET_SCENE: "GET_SCENE",
        GET_USER: "GET_USER",
        GET_GAMERS: "GET_GAMERS",
        GET_FRIENDS: "GET_FRIENDS",
        GET_INVITES: "GET_INVITES",
        LOGIN: "LOGIN",
        SIGNUP: "SIGNUP",
        LOGOUT: "LOGOUT",
        GET_MOBS: "GET_MOBS",
        GET_ERROR: "GET_ERROR",
        PLAY_MUSIC: "PLAY_MUSIC", 
        STOP_MUSIC: "STOP_MUSIC", 
        SET_MUSIC_VOLUME: "SET_MUSIC_VOLUME", 
        UPDATE_ARR_BULLET_TRAJECTORY: "UPDATE_ARR_BULLET_TRAJECTORY",
    },
    TRIGGERS: {},
};
