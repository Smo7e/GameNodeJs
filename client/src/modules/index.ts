import { TError } from "./Server/types";
import Server from "./Server/Server";
import Mediator from "./Mediator/Mediator";
import Store from "./Store/Store";

export type { TError };
export { Server, Mediator, Store };
