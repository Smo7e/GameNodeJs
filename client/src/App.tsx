import React, { useState } from "react";
import { HOST, MEDIATOR } from "./config";
import { Server, Mediator, Store, TError } from "./modules";
import SignUp from "./component/SignUp/SignUp";
import Game from "./modules/Game/Game";
import Menu from "./component/Menu/Menu";
import Login from "./component/Login/Login";
import Heroes from "./component/Heroes/Heroes";
import Interface from "./component/Interface/Interface";
import Lobby from "./component/Lobby/Lobby";
import ErrorMessage from "./modules/ErrorMessage/ErrorMessage";
import Parameters from "./component/Parameters/Parameters";
import Music from "./component/Parameters/Music";
import "./App.css";

export const ServerContext = React.createContext<Server>(null!);
export const MediatorContext = React.createContext<Mediator>(null!);
export const StoreContext = React.createContext<Store>(null!);

export enum EPAGES {
    SIGNUP,
    LOGIN,
    GAME,
    MENU,
    HEROES,
    LOBBY,
    PARAMETERS,
}

const MainApp = () => {
    const [epages, setEpages] = useState<EPAGES>(EPAGES.LOGIN);
    return (
        <>
            {epages === EPAGES.LOGIN ? (
                <Login epages={setEpages} />
            ) : epages === EPAGES.SIGNUP ? (
                <SignUp epages={setEpages} />
            ) : epages === EPAGES.MENU ? (
                <Menu epages={setEpages} />
            ) : epages === EPAGES.GAME ? (
                <>
                    <Game />
                    <Interface />
                </>
            ) : epages === EPAGES.HEROES ? (
                <Heroes epages={setEpages} />
            ) : epages === EPAGES.LOBBY ? (
                <Lobby epages={setEpages} />
            ) : epages === EPAGES.PARAMETERS ? (
                <Parameters epages={setEpages} />
            ) : (
                <></>
            )}
            <ErrorMessage />
            <Music />
        </>
    );
};

const App: React.FC = () => {
    const store = new Store();
    const mediator = new Mediator(MEDIATOR);
    const server = new Server(HOST, mediator, store);

    const { SERVER_ERROR } = mediator.getEventTypes();
    mediator.subscribe(SERVER_ERROR, (data: TError) => {
        console.log(data);
    });

    return (
        <StoreContext.Provider value={store}>
            <MediatorContext.Provider value={mediator}>
                <ServerContext.Provider value={server}>
                    <MainApp />
                </ServerContext.Provider>
            </MediatorContext.Provider>
        </StoreContext.Provider>
    );
};

export default App;
