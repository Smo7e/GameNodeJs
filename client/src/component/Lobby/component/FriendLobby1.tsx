import { useContext, useState } from "react";
import { ELOBBY, EPANEL } from "../Lobby";
import { MediatorContext, StoreContext } from "../../../App";
import SportikLobby from "./SportikLobby";
import HumanitarianLobby from "./HumanitarianLobby";
import TechguyLobby from "./TechguyLobby";
import { VARIABLE } from "../../../modules/Store/Store";
interface IFriendLobby1Props {
    setPanel: Function;
    gamers: any;
}
const FriendLobby1: React.FC<IFriendLobby1Props> = ({ setPanel, gamers }) => {
    const mediator = useContext(MediatorContext);
    const store = useContext(StoreContext);

    const [lobbyFriend1, setLobbyFriend1] = useState<ELOBBY>(ELOBBY.SPORTIK);
    if (!gamers) return <></>;

    return (
        <>
            {gamers.length === 1 ? (
                <div id="test-image-rack1" className="image-rack1">
                    <div onClick={() => setPanel(EPANEL.ADDAFRIEND1)} id="test-friend" className="friend"></div>
                </div>
            ) : gamers.length > 1 && store.get(VARIABLE.GAMER).post != "Friend-1" ? (
                <>
                    {gamers && gamers[1].person_id - 0 === 0 ? (
                        <div className="image-Sportik1">
                            <button className="button">&lt;Спортик&gt;</button>
                        </div>
                    ) : gamers && gamers[1].person_id - 0 === 1 ? (
                        <div className="image-techguy1">
                            <button className="button">&lt;Технарь&gt;</button>
                        </div>
                    ) : gamers && gamers[1].person_id - 0 === 2 ? (
                        <div className="image-humanitarian1">
                            <button className="button">&lt;Гуманитарий&gt;</button>
                        </div>
                    ) : (
                        <></>
                    )}
                </>
            ) : lobbyFriend1 === ELOBBY.SPORTIK ? (
                <SportikLobby lobby={setLobbyFriend1} gamerNumber={1} />
            ) : lobbyFriend1 === ELOBBY.HUMANITARIAN ? (
                <HumanitarianLobby lobby={setLobbyFriend1} gamerNumber={1} />
            ) : lobbyFriend1 === ELOBBY.TECHGUY ? (
                <TechguyLobby lobby={setLobbyFriend1} gamerNumber={1} />
            ) : (
                <div id="test-image-rack1" className="image-rack1"></div>
            )}
        </>
    );
};
export default FriendLobby1;
