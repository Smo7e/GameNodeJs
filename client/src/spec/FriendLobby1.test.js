import { render, screen } from "@testing-library/react";
import FriendLobby1 from "../component/Lobby/component/FriendLobby1";
import { MediatorContext, ServerContext } from "../App";
import { Mediator, Server, Store } from "../modules";
import { HOST, MEDIATOR } from "../config";

const store = new Store();
const mediator = new Mediator(MEDIATOR);
const server = new Server(HOST, mediator, store);



// test("FriendLobby1 snapshot", () => {
//     const { asFragment } = render(
//         <MediatorContext.Provider value={mediator}>
//             <ServerContext.Provider value={server}>
//                 {/* <FriendLobby1 setPanel={() => {}} gamers /> */}
//             </ServerContext.Provider>
//         </MediatorContext.Provider>
//     );
//     expect(asFragment()).toMatchSnapshot();
// });
describe('FriendLobby1 component', () => {
    test('renders correctly with Admin post and gamers length <= 2', () => {
        const { asFragment } = render(
                    <MediatorContext.Provider value={mediator}>
                        <ServerContext.Provider value={server}>
                            <FriendLobby1 setPanel={() => {}} gamers={[]}  />
                        </ServerContext.Provider>
                    </MediatorContext.Provider>
                );
                expect(asFragment()).toMatchSnapshot();
    });
    test("FriendLobby1 snapshot gamerNumber 0", () => {
        const gamers = [{ person_id: 0 }, { person_id: 1 }, { person_id: 2 }];
    const { asFragment } = render(
        <MediatorContext.Provider value={mediator}>
            <ServerContext.Provider value={server}>
                <FriendLobby1 setPanel={() => {}} gamers={[]}  />
            </ServerContext.Provider>
        </MediatorContext.Provider>
    );
    expect(asFragment()).toMatchSnapshot();
});
test("FriendLobby1 snapshot gamerNumber 1", () => {
    const gamers = [{ person_id: 0 }, { person_id: 1 }];
    const { asFragment } = render(
        <MediatorContext.Provider value={mediator}>
            <ServerContext.Provider value={server}>
                <FriendLobby1 setPanel={() => {}} gamers={[]}  />
            </ServerContext.Provider>
        </MediatorContext.Provider>
    );
    expect(asFragment()).toMatchSnapshot();
});

test("FriendLobby1 snapshot gamerNumber 2", () => {
    const gamers = [{ person_id: 0 }, { person_id: 1 }, { person_id: 3 }];
    const { asFragment } = render(
        <MediatorContext.Provider value={mediator}>
            <ServerContext.Provider value={server}>
                <FriendLobby1 setPanel={() => {}} gamers={[]}  />
            </ServerContext.Provider>
        </MediatorContext.Provider>
    );
    expect(asFragment()).toMatchSnapshot();
});

});