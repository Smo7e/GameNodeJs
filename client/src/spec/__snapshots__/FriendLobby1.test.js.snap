// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FriendLobby1 snapshot 1`] = `<DocumentFragment />`;
